### ISSUE TITLE
Make 0.7 tilesets external

### ISSUE THREAD

**Issue by AssassinTee** [ddnet/ddnet#5164]
Currently when you add for example `generic_unhookable_0.7` it is always embedded and can't be turned external:

![screenshot_2022-05-20_11-09-39](https://user-images.githubusercontent.com/821401/169495553-bf2f7d5a-bf1e-442d-990a-6c6ebe9857aa.png)

Shouldn't they be added to the list of vanilla tilsets and be external? https://github.com/ddnet/ddnet/blob/61cca6ca16611fd2a3c4e94fbd68dc8ab4931e05/src/game/editor/editor.cpp#L40

This would allow for smaller maps, because the tileset doesn't need to be embedded into the map. Also every ddnet player should have it, because it's in the data https://github.com/ddnet/ddnet/blob/master/data/mapres/generic_unhookable_0.7.png

If you are sending this map to a 0.7 client, this mapres should be renamed to `generic_unhookable`. Other mapres, which were external in 0.6, will still break in 0.7 anyway. For example `winter_main` does not have a `winter_main_0.7`

**Comment by heinrich5991** [comment]
It's not in there because it's not a vanilla 0.6 image and as such, can't be assumed to exist.

To fix the general issue of map sizes, it'd probably be best to implement a generic image downloading process.

**Comment by AssassinTee** [comment]
> It's not in there because it's not a vanilla 0.6 image and as such, can't be assumed to exist.

It can be assumed to exist for ddnet-players and for 0.7 players (while having a different name) (but not for 0.6 players).

Aren't you sending different versions of a map to different clients anyways due to compatiblity?

> To fix the general issue of map sizes, it'd probably be best to implement a generic image downloading process.

That really would be nice and could be done like the skin downloads. However this would probably bring a lot of it's own problems with it

**Comment by heinrich5991** [comment]
> Aren't you sending different versions of a map to different clients anyways due to compatiblity?

No, only for 0.7, and we're simply embedding all 0.6 standard images to achieve compatibility.

**Comment by C0D3D3V** [comment]
>    To fix the general issue of map sizes, it'd probably be best to implement a generic image downloading process.

would probably cause that we would have a 3rd map format xD for clients who can download the images via api. The clients would have to be able to disclose that they are able to use the api (or we assume as usual that new versions can), and then the question is from where are the images requested from an extra server hosted by ddnet or from the game server itself.

would the server then extract the images from the map or would the images be kept separately xD pretty elaborate issue, should probably open an extra issue for it since it does not quite fit the actual topic.

**Comment by AssassinTee** [comment]
> No, only for 0.7, and we're simply embedding all 0.6 standard images to achieve compatibility.

Uff, while this is working there is a lot of optimization potential. Most tilesets didn't change from 0.6 to 0.7 and could stay external.

@heinrich5991 thank you for the discussion, I think I opened a can of worms and we all know: [Premature Optimization Is the Root of All Evil](https://stackify.com/premature-optimization-evil/)

**Comment by AssassinTee** [comment]
> would the server then extract the images from the map or would the images be kept separately xD pretty elaborate issue, should probably open an extra issue for it since it does not quite fit the actual topic.

Reasons for the map to contain the image and for the server to send it:
- scability, it's more decentralized, one image server would be a huge target
- mapres name is not sufficient for uniqueness
  - you can have various mapres with the same name which are different, probably requries image hashes
  - versioning? lol

**Comment by heinrich5991** [comment]
> oops

See #5165. I addressed scalability and uniqueness.
